src/motioncore/protocols/boolean_gmw/boolean_gmw_gate.cpp

```diff
@@ -764,4 +764,96 @@ const motion::SharePointer MuxGate::GetOutputAsShare() const {
   return result;
 }
 
+ConstantAsBooleanGmwInputGate::ConstantAsBooleanGmwInputGate(std::span<const BitVector<>> input,
+                                                             Backend& backend)
+    : ConstantAsBooleanGmwInputGate::Base(backend),
+      input_(std::vector(input.begin(), input.end())) {
+  input_owner_id_ = 0;
+  bits_ = input_.size() == 0 ? 0 : input_.at(0).GetSize();
+  InitializationHelper();
+}
+
+ConstantAsBooleanGmwInputGate::ConstantAsBooleanGmwInputGate(std::vector<BitVector<>>&& input,
+                                                             Backend& backend)
+    : ConstantAsBooleanGmwInputGate::Base(backend), input_(std::move(input)) {
+  input_owner_id_ = 0;
+  bits_ = input_.size() == 0 ? 0 : input_.at(0).GetSize();
+  InitializationHelper();
+}
+
+void ConstantAsBooleanGmwInputGate::InitializationHelper() {
+  auto& communication_layer = GetCommunicationLayer();
+  auto& _register = GetRegister();
+
+  if (static_cast<std::size_t>(input_owner_id_) >= communication_layer.GetNumberOfParties()) {
+    throw std::runtime_error(fmt::format("Invalid input owner: {} of {}", input_owner_id_,
+                                         communication_layer.GetNumberOfParties()));
+  }
+
+  gate_id_ = _register.NextGateId();
```

Comment:
`gate_id_` is not assigned in a specific gate's constructor anymore. Please remove it.

Reply:
Line 793 (gate_id_ = _register.NextGateId();) is already removed.